ci/test/00_setup_env_native_asan.sh

```diff
@@ -7,7 +7,7 @@
 export LC_ALL=C.UTF-8
 
 export CONTAINER_NAME=ci_native_asan
-export PACKAGES="clang llvm python3-zmq qtbase5-dev qttools5-dev-tools libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-test-dev libdb5.3++-dev libminiupnpc-dev libnatpmp-dev libzmq3-dev libqrencode-dev libsqlite3-dev"
+export PACKAGES="clang llvm python3-zmq qtbase5-dev qttools5-dev-tools libevent-dev bsdmainutils libboost-dev libboost-system-dev libboost-filesystem-dev libboost-test-dev libdb5.3++-dev libminiupnpc-dev libnatpmp-dev libzmq3-dev libqrencode-dev libsqlite3-dev"
```

Comment:
Isn't `libboost-dev` an umbrella package forl *all* of boost?

Reply:
Yes. Note that it's currently already being installed, as a side-effect of either `libbost-filesystem-dev` or `libboost-system-dev`. However, once we are no-longer installing those packages, i.e post #20744, users will still need to install the boost headers, for multi-index, process etc. As far as I can, tell, `libboost-dev` is the leanest way to achieve that. There is no `libbbost-muti-index-dev`, or `libboost-process-dev` packages or similar from what I can find, that we could selectively use to install just the components we want.

In a clean Ubuntu Focal vm if you compare `libboost-dev` to `libboost-all-dev`, the difference is quite substantial:

### apt install libboost-dev
```bash
The following NEW packages will be installed:
  gcc-7-base gcc-8-base libasan4 libatomic1 libboost-dev libboost1.71-dev libc-dev-bin libc6-dev libcilkrts5 libcrypt-dev libgcc-7-dev libgomp1
  libitm1 liblsan0 libmpx2 libquadmath0 libstdc++-7-dev libtsan0 libubsan0 linux-libc-dev manpages manpages-dev
```

### apt install libboost-all-dev
```bash
The following NEW packages will be installed:
  autoconf automake autotools-dev binutils binutils-common binutils-x86-64-linux-gnu cpp cpp-8 cpp-9 file gcc gcc-7-base gcc-8 gcc-8-base gcc-9
  gcc-9-base gfortran gfortran-8 gfortran-9 ibverbs-providers icu-devtools krb5-locales libasan4 libasan5 libatomic1 libbinutils libboost-all-dev
  libboost-atomic-dev libboost-atomic1.71-dev libboost-atomic1.71.0 libboost-chrono-dev libboost-chrono1.71-dev libboost-chrono1.71.0
  libboost-container-dev libboost-container1.71-dev libboost-container1.71.0 libboost-context-dev libboost-context1.71-dev libboost-context1.71.0
  libboost-coroutine-dev libboost-coroutine1.71-dev libboost-coroutine1.71.0 libboost-date-time-dev libboost-date-time1.71-dev
  libboost-date-time1.71.0 libboost-dev libboost-exception-dev libboost-exception1.71-dev libboost-fiber-dev libboost-fiber1.71-dev
  libboost-fiber1.71.0 libboost-filesystem-dev libboost-filesystem1.71-dev libboost-filesystem1.71.0 libboost-graph-dev
  libboost-graph-parallel-dev libboost-graph-parallel1.71-dev libboost-graph-parallel1.71.0 libboost-graph1.71-dev libboost-graph1.71.0
  libboost-iostreams-dev libboost-iostreams1.71-dev libboost-iostreams1.71.0 libboost-locale-dev libboost-locale1.71-dev libboost-locale1.71.0
  libboost-log-dev libboost-log1.71-dev libboost-log1.71.0 libboost-math-dev libboost-math1.71-dev libboost-math1.71.0 libboost-mpi-dev
  libboost-mpi-python-dev libboost-mpi-python1.71-dev libboost-mpi-python1.71.0 libboost-mpi1.71-dev libboost-mpi1.71.0 libboost-numpy-dev
  libboost-numpy1.71-dev libboost-numpy1.71.0 libboost-program-options-dev libboost-program-options1.71-dev libboost-program-options1.71.0
  libboost-python-dev libboost-python1.71-dev libboost-python1.71.0 libboost-random-dev libboost-random1.71-dev libboost-random1.71.0
  libboost-regex-dev libboost-regex1.71-dev libboost-regex1.71.0 libboost-serialization-dev libboost-serialization1.71-dev
  libboost-serialization1.71.0 libboost-stacktrace-dev libboost-stacktrace1.71-dev libboost-stacktrace1.71.0 libboost-system-dev
  libboost-system1.71-dev libboost-system1.71.0 libboost-test-dev libboost-test1.71-dev libboost-test1.71.0 libboost-thread-dev
  libboost-thread1.71-dev libboost-thread1.71.0 libboost-timer-dev libboost-timer1.71-dev libboost-timer1.71.0 libboost-tools-dev
  libboost-type-erasure-dev libboost-type-erasure1.71-dev libboost-type-erasure1.71.0 libboost-wave-dev libboost-wave1.71-dev libboost-wave1.71.0
  libboost1.71-dev libboost1.71-tools-dev libbsd0 libc-dev-bin libc6-dev libcaf-openmpi-3 libcbor0.6 libcc1-0 libcilkrts5 libcoarrays-dev
  libcoarrays-openmpi-dev libcrypt-dev libctf-nobfd0 libctf0 libedit2 libevent-2.1-7 libevent-core-2.1-7 libevent-dev libevent-extra-2.1-7
  libevent-openssl-2.1-7 libevent-pthreads-2.1-7 libexpat1 libexpat1-dev libfabric1 libfido2-1 libgcc-7-dev libgcc-8-dev libgcc-9-dev
  libgdbm-compat4 libgdbm6 libgfortran-8-dev libgfortran-9-dev libgfortran5 libgomp1 libgssapi-krb5-2 libhwloc-dev libhwloc-plugins libhwloc15
  libibverbs-dev libibverbs1 libicu-dev libicu66 libisl22 libitm1 libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0 liblsan0 libltdl-dev
  libltdl7 libmagic-mgc libmagic1 libmpc3 libmpdec2 libmpfr6 libmpx2 libnl-3-200 libnl-3-dev libnl-route-3-200 libnl-route-3-dev libnuma-dev
  libnuma1 libopenmpi-dev libopenmpi3 libpciaccess0 libperl5.30 libpmix2 libpsm-infinipath1 libpsm2-2 libpython3-dev libpython3-stdlib
  libpython3.8 libpython3.8-dev libpython3.8-minimal libpython3.8-stdlib libquadmath0 librdmacm1 libreadline8 libsigsegv2 libsqlite3-0 libssl1.1
  libstdc++-7-dev libtool libtsan0 libubsan0 libubsan1 libx11-6 libx11-data libxau6 libxcb1 libxdmcp6 libxext6 libxml2 libxmuu1 libxnvctrl0
  linux-libc-dev m4 manpages manpages-dev mime-support mpi-default-bin mpi-default-dev netbase ocl-icd-libopencl1 openmpi-bin openmpi-common
  openssh-client perl perl-modules-5.30 python3 python3-dev python3-distutils python3-lib2to3 python3-minimal python3.8 python3.8-dev
  python3.8-minimal readline-common tzdata xauth xz-utils zlib1g-dev
```

Reply:
I guess that makes the answer to my question 'no' — `libboost-all-dev` is the umbrella package that I meant. This is okay.